definitions/npm/react-apollo_v2.x.x/flow_v0.58.x-/react-apollo_v2.x.x.js

```diff
@@ -211,25 +211,43 @@ declare module 'react-apollo' {
 
   declare export function cleanupApolloState(apolloState: any): void;
 
-  declare export type QueryRenderProps<TData, TVariables> = {
+  declare export type SubscribeToMoreOptions<TData, TSubscriptionData, TSubscriptionVariables=void> = {
+    document?: DocumentNode,
+    variables?: TSubscriptionVariables,
+    updateQuery?: (previousResult: TData, result: {subscriptionData: {data?: TSubscriptionData}, variables: TSubscriptionVariables}) => TData,
+    onError?: Function
+  }
+
+  declare type FetchMoreOptions<TData, TVariables> = {|
+    variables?: TVariables,
+    updateQuery: (previousResult: TData, {fetchMoreResult: TData, variables: TVariables}) => TData
+  |}
+
+  declare type FetchMoreQueryOptions<TData, TVariables, TFetchMoreData, TFetchMoreVariables=void> = {|
+    query: DocumentNode,
+    variables?: TFetchMoreVariables,
+    updateQuery: (previousResult: TData, {fetchMoreResult: TFetchMoreData, variables: TFetchMoreVariables}) => TData
+  |}
+
+  declare export type QueryRenderProps<TData, TVariables=void> = {
     data?: TData | {||},
     loading: boolean,
     error?: ApolloError,
     variables: TVariables,
     networkStatus: NetworkStatus,
     refetch: (variables?: TVariables) => Promise<mixed>,
-    fetchMore: ({query?: DocumentNode, variables?: TVariables, updateQuery: Function}) => Promise<mixed>,
+    fetchMore: (options: FetchMoreOptions<TData, TVariables> | FetchMoreQueryOptions<TData, TVariables, any, any>) => Promise<mixed>,
     load: () => void,
     startPolling: (interval: number) => void,
     stopPolling: (interval: number) => void,
-    subscribeToMore: (options: {document?: DocumentNode, variables?: TVariables, updateQuery?: Function, onError?: Function}) => () => void,
-    updateQuery: (previousResult: TData, options?: {variables: TVariables}) => TData,
+    subscribeToMore: (options: SubscribeToMoreOptions<TData, any, any>) => () => void,
+    updateQuery: (previousResult: TData, options: {variables: TVariables}) => TData,
     client: ApolloClient
   }
 
   declare export type QueryRenderPropFunction<TData, TVariables> = (QueryRenderProps<TData, TVariables>) => React$Node
 
-  declare export class Query<TData, TVariables> extends React$Component<{
+  declare export class Query<TData, TVariables, TSubscriptionData, TSubscriptionVariables> extends React$Component<{
```

Comment:
Would it make sense to add defaults for these?

At the moment, I'm extending the Query component and passing in the types, similar to how the apollo typescript docs suggest (https://www.apollographql.com/docs/react/recipes/static-typing.html#typing-components). This change would require going back and adding void, void for each of my Query components.

Reply:
Actually adding these was a mistake and they aren't used. I chose not to propagate these type variables because their usability is doubtable. The Query component doesn't "care" about those types, and that is also why the `QueryRenderProps` doesn't have them.